### ISSUE TITLE
Aligning with `kotlin-dsl` kotlin version

### ISSUE THREAD

**Issue by Wrywulf** [ReactiveCircus/app-versioning#5]
Hi, 

Firstly, thanks for the work on this plugin! My company applies a similar solution (git tag / semver) to our projects but in a more rudimentary way. So, looking forward to adopting this plugin and the lazy configuration practices.

My use case is to apply this plugin to a bunch of application modules in my repo. They should all follow our internal versioning practices, which are natural extensions of what this plugin can offer.

I'm building an opinionated "wrapper" plugin (buildSrc) around app-versioning that implements  these internal practices. App modules should then apply the wrapper plugin instead.

When developing the plugin I noticed that I could not use the types from app-versioning due to this error:

```
e: Classes compiled by a new Kotlin compiler backend were found in dependencies. Remove them from the classpath or use '-Xallow-jvm-ir-dependencies' to suppress errors
..
Class 'io.github.reactivecircus.appversioning.AppVersioningExtension' is compiled by a new Kotlin compiler backend and cannot be loaded by the old compiler
```

It seems that the `kotlin-dsl` plugin internally still uses kotlin 1.3.72.
I see from the source that app-versioning applies 1.4.20-M2 at the moment.

Would it make sense to stay at 1.3.72 (for now) to avoid these kind of errors?

Also considering that AGP does the same: https://twitter.com/droidxav/status/1321933969135345666?s=20

I guess I could apply the '-Xallow-jvm-ir-dependencies' to suppress errors but I would like to avoid these workarounds as much as possible.

**Comment by ychescale9** [comment]
Hi, thanks for trying the plugin!

I've been pretty aggressive with updating dependencies (e.g. kotlin, AGP) since version 0.4.0 because the new AGP variant API that this plugin depends on is incubating and has made some binary incompatible changes recently, so I basically gave up supporting the plugin for stable AGP (4.0, 4.1).

The error you're getting however is likely due to the IR backend that I've enabled (for no practical reason).

We can fix this by disabling IR, but just want to check a few things before shipping the fix:

- Are you using 0.5.0 of the plugin (and latest version of AGP)?
- what's the kotlin version used your `buildSrc` classpath?

**Comment by Wrywulf** [comment]
Thanks for the quick response.

> Are you using 0.5.0 of the plugin (and latest version of AGP)?
I am using 0.4.0 for now. But plan on stepping up to 0.5+ once AGP 4.2 is beta.

> what's the kotlin version used your buildSrc classpath?
I only specify the `kotlin-dsl` plugin
plugins {
    `kotlin-dsl`
}
So assuming that it is implicitly resolved by that to 1.3.72.
When I tried to explicitly add kotlin("jvm") version "1.4.10" it gives me this warning:
```
> Configure project :buildSrc
WARNING: Unsupported Kotlin plugin version.
The `embedded-kotlin` and `kotlin-dsl` plugins rely on features of Kotlin `1.3.72` that might work differently than in the requested version `1.4.10`.
```
Does that answer your question?
Not super familiar with the gradle internals yet, so apologize If I misunderstood the question.

**Comment by ychescale9** [comment]
`WARNING: Unsupported Kotlin plugin version.` should be ok which any plugins using kotlin 1.4 are getting right now.

The compilation error you mentioned is earlier is a legit issue though as we can't expect `kotlin-dsl` to be compiling with IR anytime soon.

I was able to reproduce the error locally with a project using `kotlin-dsl` and kotlin gradle plugin `1.4.10` in `buildSrc`, and disabling IR fixes the issue.

Shipping the fix on top of `0.5.0` is a bit tricky and since AGP 4.2 beta should be relatively close, I'll probably fix it and release 0.6.0 once AGP 4.2-beta is out.

Thanks again for trying out this plugin early!
